instrumentation/http/src/test/java/brave/http/features/secondary_sampling/SecondarySamplingIntegratedTest.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.http.features.secondary_sampling;
+
+import brave.Tracing;
+import brave.TracingCustomizer;
+import brave.propagation.B3SinglePropagation;
+import brave.propagation.Propagation;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import org.junit.After;
+import org.junit.Test;
+import zipkin2.Callback;
+import zipkin2.DependencyLink;
+import zipkin2.reporter.Reporter;
+import zipkin2.storage.InMemoryStorage;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * This is a <a href="https://github.com/openzipkin/openzipkin.github.io/wiki/secondary-sampling">Secondary
+ * Sampling</a> proof of concept.
+ */
+public class SecondarySamplingIntegratedTest {
```

Comment:
this is now converged, one scenario (skipped hops in gatewayplay) doesn't work yet